src/scripts/clipperUI/clipper.tsx

```diff
@@ -620,9 +621,22 @@ class ClipperClass extends ComponentBase<ClipperState, {}> {
 
 		this.state.setState({ oneNoteApiResult: { status: Status.InProgress } });
 		SaveToOneNote.startClip(this.state).then((startClipPackage: StartClipPackage) => {
-			clipEvent.setCustomProperty(Log.PropertyName.Custom.CorrelationId, startClipPackage.responsePackage.request.getResponseHeader(Constants.HeaderValues.correlationId));
-			clipEvent.setCustomProperty(Log.PropertyName.Custom.AnnotationAdded, startClipPackage.annotationAdded);
-			this.state.setState({ oneNoteApiResult: { data: startClipPackage.responsePackage.parsedResponse, status: Status.Succeeded } });
+			RatingsHelper.incrementClipSuccessCount().then(() => {
+				// do nothing special on incrementClipSuccessCount resolve
+			}).then(() => {
+				// will happen regardless of success/failure of incrementClipSuccessCount
```

Comment:
Yeah, I just want to be extra careful that `this.state.setState({ oneNoteApiResult: { data: startClipPackage.responsePackage.parsedResponse, status: Status.Succeeded } });` isn't ever not set because we (for example) tried to read a property off an `undefined` value inside the promise, or something - that's now with this implementation, but also if anyone updates this logical flow later on.